lib/gcr/cassette.rb

```diff
@@ -77,46 +77,56 @@ def play(&blk)
   end
 
   def start_recording
-    GCR.stub.class.class_eval do
-      alias_method :orig_request_response, :request_response
-
-      def request_response(*args)
-        orig_request_response(*args).tap do |resp|
-          req = GCR::Request.from_proto(*args)
-          if GCR.cassette.reqs.none? { |r, _| r == req }
-            GCR.cassette.reqs << [req, GCR::Response.from_proto(resp)]
+    GCR.stubs.each do |klass|
+      klass.class_eval do
+        alias_method :orig_request_response, :request_response
+
+        def request_response(*args)
+          orig_request_response(*args).tap do |resp|
+            req = GCR::Request.from_proto(*args)
+            if GCR.cassette.reqs.none? { |r, _| r == req }
+              GCR.cassette.reqs << [req, GCR::Response.from_proto(resp)]
+            end
           end
         end
-      end
+      end unless klass.respond_to?(:orig_request_response)
     end
   end
 
   def stop_recording
-    GCR.stub.class.class_eval do
-      alias_method :request_response, :orig_request_response
+    GCR.stubs.each do |klass|
+      klass.class_eval do
+        alias_method :request_response, :orig_request_response
+        undef :orig_request_response
+      end if klass.respond_to?(:orig_request_response)
     end
     save
   end
 
   def start_playing
     load
 
-    GCR.stub.class.class_eval do
-      alias_method :orig_request_response, :request_response
+    GCR.stubs.each do |klass|
+      klass.class_eval do
+        alias_method :orig_request_response, :request_response
 
-      def request_response(*args)
-        request_proto = args[1]
-        GCR.cassette.reqs.each do |other_req, resp|
-          return resp.to_proto if request_proto.to_s == other_req.to_proto.last.to_s
+        def request_response(*args)
+          request_proto = args[1]
+          GCR.cassette.reqs.each do |other_req, resp|
+            return resp.to_proto if request_proto.to_s == other_req.to_proto.last.to_s
+          end
+          raise GCR::NoRecording
         end
-        raise GCR::NoRecording
-      end
+      end if !klass.respond_to?(:orig_request_response)
     end
   end
 
   def stop_playing
-    GCR.stub.class.class_eval do
-      alias_method :request_response, :orig_request_response
+    GCR.stubs.each do |klass|
+      klass.class_eval do
+        alias_method :request_response, :orig_request_response
+        undef :orig_request_response
```

Comment:
Curious why do you need to `undef :orig_request_response`?

Reply:
After thinking about it, the patch would be a little simpler without the `undef`, so I have removed.  Originally, I was thinking of it as a matter of cleanup/symmetry (since `cassette` defined the method, it also removed it). I did some local testing without the `undef` and code seemed to behave appropriately.

Reply:
👍